lib/services/access_checker.go

```diff
@@ -1009,6 +1009,9 @@ type HostUsersInfo struct {
 	GID string
 	// Shell is the default login shell for a host user
 	Shell string
+	// TakeOwnership determines whether or not an existing user should be
+	// taken over by teleport
+	TakeOwnership bool
```

Comment:
I think we should eventually support this on the role spec as well and consider removing the implicit migration when `teleport-keep` is present in a role's `host_groups`.